src/components/admin/CounterConfig.js

```diff
@@ -0,0 +1,185 @@
+import React, {Component} from 'react';
+import {
+    Button,
+    Input,
+    InputGroup
+} from "reactstrap";
+
+import moment from "moment";
+
+import DB from "../../state/DB";
+
+export default class CounterConfig extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            hour_counters: {},
+            day_counters: {}
+        };
+
+        this.isConfigValid = this.isConfigValid.bind(this);
+        this.handleClick = this.handleClick.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    componentWillMount() {
+        const config = DB.config;
+
+        const format = key => Object.keys(config[key]).reduce((acc, cur) => {
+            acc[cur] = false;
+            return acc;
+        }, {});
+
+        let hour_counters = format("hour_counters");
+        let day_counters = format("day_counters");
+
+        this.setState({
+            config: config,
+            hour_counters: hour_counters,
+            day_counters: day_counters,
+        });
+    }
+
+    isConfigValid() {
+        return Object.values({...this.state.config.hour_counters, ...this.state.config.day_counters}).every(
+            counter => counter.constructor === Array ? counter.every(date => moment(date).isValid()) : typeof counter === "number")
+    }
+
+    handleClick(type, counter) {
+        this.setState({
+            [type]: {
+                ...this.state[type],
+                [counter]: !this.state[type][counter]
+            }
+        })
+    }
+
+    handleChange(event, counter_type, index) {
+        let obj = {
+            [counter_type]: this.state.config[counter_type]
+        };
+
+        if (index === undefined) {
+            obj[counter_type][event.target.name] = parseInt(event.target.value) || 0;
+        } else {
+            obj[counter_type][event.target.name] = [...this.state.config[counter_type][event.target.name]];
+            obj[counter_type][event.target.name][index] = event.target.value;
+        }
+
+        this.setState({
+            config: {
+                ...this.state.config,
+                ...obj
+            }
+        })
+    }
+
+    handleSubmit(event) {
+        event.preventDefault();
+        if (this.isConfigValid()) {
+            DB.setConfig(this.state.config);
+            DB.updateConfigFile();
+        }
+    }
+
+    render() {
+        return (
+            <div>
```

Comment:
consider maybe using `React.Fragment`, an array or just `<>` for these grouping divs